platform/darwin/src/MGLThreadPool.mm

```diff
@@ -0,0 +1,39 @@
+#import "MGLThreadPool_Private.h"
+
+#import "NSProcessInfo+MGLAdditions.h"
+
+@interface MGLThreadPool ()
+
+@property (nonatomic) mbgl::ThreadPool *mbglThreadPool;
+
+@end
+
+@implementation MGLThreadPool
+
+#pragma mark - Internal
+
++ (instancetype)sharedPool {
+    if (NSProcessInfo.processInfo.mgl_isInterfaceBuilderDesignablesAgent) {
```

Comment:
This is unnecessary, since there won't be an offline storage object if we're rendering for IB, and even if we were, spinning up this thread pool should be cheap. The reason we do this in MGLOfflineStorage and MGLAccountManager is to avoid potentially expensive operations that may block MGLMapView from rendering within the allotted time.